logger/syslogd/syslogd.go

```diff
@@ -106,18 +110,24 @@ func Listen(signalChan chan os.Signal, cleanupDone chan bool, bindAddr string) {
 			log.Fatalf("unable to create LogRoot at %s: %v", LogRoot, err)
 		}
 	}
-	// Create a server with one handler and run one listen gorutine
+	// Create a server with one handler and run one listen goroutine
 	s := syslog.NewServer()
-	s.AddHandler(newHandler())
+	h := newHandler()
+	s.AddHandler(h)
 	s.Listen(bindAddr)
 	fmt.Println("Syslog server started...")
 	fmt.Println("deis-logger running")
 
 	// Wait for terminating signal
-	for _ = range signalChan {
-		// Shutdown the server
-		fmt.Println("Shutting down...")
-		s.Shutdown()
-		cleanupDone <- true
+	for {
+		select {
+		case <-exitChan:
+			// Shutdown the server
+			fmt.Println("Shutting down...")
+			s.Shutdown()
+			cleanupDone <- true
+		case d := <-drainChan:
+			h.drainURI = d
```

Comment:
could you please add a test or at the least a TODO to add tests for this? We'd want to make sure that the drain URI is updated when we send some new value over the channel.